apm-agent-core/src/main/java/co/elastic/apm/bci/bytebuddy/MinimumClassFileVersionValidator.java

```diff
@@ -0,0 +1,61 @@
+package co.elastic.apm.bci.bytebuddy;
+
+import net.bytebuddy.ClassFileVersion;
+import net.bytebuddy.asm.AsmVisitorWrapper;
+import net.bytebuddy.description.field.FieldDescription;
+import net.bytebuddy.description.field.FieldList;
+import net.bytebuddy.description.method.MethodList;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.Implementation;
+import net.bytebuddy.jar.asm.ClassVisitor;
+import net.bytebuddy.pool.TypePool;
+import net.bytebuddy.utility.OpenedClassReader;
+
+public enum MinimumClassFileVersionValidator implements AsmVisitorWrapper {
+
+    INSTANCE;
+
+    private static final ClassFileVersion MINIMUM_CLASS_FILE_VERSION = ClassFileVersion.JAVA_V5;
+
+    @Override
+    public ClassVisitor wrap(TypeDescription instrumentedType, ClassVisitor classVisitor, Implementation.Context implementationContext,
```

Comment:
We talked about switching the order of the calls to `transform`, but maybe what you did is the only right way to go if this is expected to wrap the `ForAdvice` `ClassVisitor`